cli/upgrade.rs

```diff
@@ -0,0 +1,210 @@
+// Copyright 2018-2020 the Deno authors. All rights reserved. MIT license.
+
+//! This module provides feature to upgrade deno executable
+//!
+//! At the moment it is only consumed using CLI but in
+//! the future it can be easily extended to provide
+//! the same functions as ops available in JS runtime.
+
+extern crate semver_parser;
+use crate::futures::FutureExt;
+use crate::{
+  http_util::{fetch_once, FetchOnceResult},
+  ErrBox,
+};
+use regex::Regex;
+use reqwest::{redirect::Policy, Client};
+use semver_parser::version::parse as semver_parse;
+use semver_parser::version::Version;
+use std::future::Future;
+use std::io::prelude::*;
+use std::path::Path;
+use std::path::PathBuf;
+use std::pin::Pin;
+use std::process::Command;
+use std::string::String;
+use tempfile::TempDir;
+use url::Url;
+
+// TODO(ry) we should really be using target triples for the uploaded files.
+#[cfg(windows)]
+const EXEC_FILE_NAME: &str = "deno_win_x64.zip";
+#[cfg(target_os = "macos")]
+const EXEC_FILE_NAME: &str = "deno_osx_x64.gz";
+#[cfg(target_os = "linux")]
+const EXEC_FILE_NAME: &str = "deno_linux_x64.gz";
+
+#[cfg(windows)]
+const UNZIP_COMMAND: &str = "Expand-Archive";
+#[cfg(windows)]
+const EXT: &str = "exe";
+#[cfg(not(windows))]
+const UNZIP_COMMAND: &str = "gunzip";
+#[cfg(not(windows))]
+const EXT: &str = "";
```

Comment:
It's not necessary to make these global constants. It creates unnecessary indirection - just use string literals in the code.